### PR TITLE
Codespell ci

### LINKED ISSUE #1236: Adding spellcheck to the CI
Similar to https://github.com/open-telemetry/opentelemetry-python/pull/2878, for the contrib repo

**Is your feature request related to a problem?**
Yes, would like to ensure spellcheck for the repo

**Describe the solution you'd like**
I would like `codespell` to run as part of the CI

**Describe alternatives you've considered**
None, based it off the fact that the opentelemetry-python repo already uses it

### PR DESCRIPTION
**Pull request by galbash**
# Description
Fixes #1236. Adding spellcheck on all files as part of the tox phase in the build. Additionally, fixed all discovered typos so the CI will pass.

## Type of change
Please delete options that are not relevant.
- [X] Chore (codebase fix that does not change the behavior in any way)
- [X] This change requires a documentation update (internal)

# How Has This Been Tested?
This PR essentially adds a test. To run spellcheck run `tox -e spellcheck`

# Does This PR Require a Contrib Repo Change?
- [ ] Yes. - Link to PR: 
- [X] No.

# Checklist:

- [X] Followed the style guidelines of this project
- [ ] Changelogs have been updated - no need
- [ ] Unit tests have been added - no need
- [X] Documentation has been updated - the internal one, in `CONTRIBUTING.md`

### CODE REVIEW

**Comment by galbash** [comment]
broken down to two commits to enable easy review:
1. adding codespell to the CI
2. fixing typos

File: .codespellrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[codespell]
+# skipping auto generated folders
+skip = ./.tox,./.mypy_cache,./docs/_build,./target
+ignore-words-list = ot,soket
```
Comment (galbash) [inline .codespellrc:4]: added `soket` since I didn't want to change the code without discussing with approvers first just for the sake of spellcheck

Comment (srikanthccv) [reply .codespellrc:4]: It's fine. You can update it's refs.

**Review by lzchen — APPROVED** [review]

File: .codespellrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[codespell]
+# skipping auto generated folders
+skip = ./.tox,./.mypy_cache,./docs/_build,./target
```
Comment (srikanthccv) [inline .codespellrc:3]:
```suggestion
skip = ./.tox,./.mypy_cache,./docs/_build,./target,*/LICENSE,./venv
```

**Review by ocelotl — APPROVED** [review]

**Comment by galbash** [comment]
fixed linting

**Pull request merged**